src/proxy/query.go

```diff
@@ -224,9 +224,17 @@ func (spanner *Spanner) ComQuery(session *driver.Session, query string, callback
 						log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
 					}
 				} else {
-					// e.g.: select a from table [as] aliasTable;
-					if qr, err = spanner.handleSelect(session, query, node); err != nil {
-						log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
+					if ok && spanner.router.IsSystemDB(tb.Qualifier.String()) {
```

Comment:
Here the logic is mesh, we should clear, i have did a patch:
```
	case *sqlparser.Select:
                ... ...
		default:
			switch snode.From[0].(type) {
			case *sqlparser.AliasedTableExpr:
				aliasTableExpr := snode.From[0].(*sqlparser.AliasedTableExpr)
				tb, ok := aliasTableExpr.Expr.(sqlparser.TableName)
				if !ok {
					// Subquery.
					if qr, err = spanner.handleSelect(session, query, node); err != nil {
						log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
					}
				} else {
					if tb.Name.String() == "dual" || spanner.router.IsSystemDB(tb.Qualifier.String()) {
						if qr, err = spanner.ExecuteSingle(query); err != nil {
							log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
						}
					} else {
						if qr, err = spanner.handleSelect(session, query, node); err != nil {
							log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
						}
					}
				}
				spanner.auditLog(session, R, xbase.SELECT, query, qr)
				return returnQuery(qr, callback, err)
			default: // ParenTableExpr, JoinTableExpr
				if qr, err = spanner.handleSelect(session, query, node); err != nil {
					log.Error("proxy.select[%s].from.session[%v].error:%+v", query, session.ID(), err)
				}
				spanner.auditLog(session, R, xbase.SELECT, query, qr)
				return returnQuery(qr, callback, err)
			}
		}
```
and remove select.go::handleDual

Reply:
Good job，I follow you.